src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -244,6 +248,9 @@ private void WatchForFileSystemEventsThreadStart(object arg)
                 _watcherRunLoop = Interop.RunLoop.CFRunLoopGetCurrent();
                 Debug.Assert(_watcherRunLoop != IntPtr.Zero);
 
+                // Retain the RunLoop so that it doesn't get moved or cleaned up before we're done with it.
+                Interop.CoreFoundation.CFRetain(_watcherRunLoop);
```

Comment:
Just to validate that it's behaving as we expect, and to help avoid confusion for folks like me that see CFRetain is defined to return an IntPtr and wonder whether we should be using it, consider changing this to something like:
```C#
IntPtr retainResult = Interop.CoreFoundation.CFRetain(_watcherRunLoop);
Debug.Assert(retainResult == _watcherRunLoop, "CFRetain is supposed to return the input value");
```